### PR TITLE
Add RabbitMQ performance testing.

### PR DESCRIPTION
**Pull request by rjweeks**
Adding the RabbitMQ Performance test tool.

To build and push image:
```
make rabbitmq-perf-test
docker push eu.gcr.io/census-rm-ci/rm/census-rm-rabbit-performance:latest
```

Deploy the container to K8S
```
kubectl apply -f rabbit-perf-test/census-rm-rabbit-perf-deployment.yml
```

Shell into pod.

Running a basic test run:
This is a continuous test so you need to 'CTRL C' to finish the test.
```
tests/basic-test.sh
```

For documentation on running tests:

https://rabbitmq.github.io/rabbitmq-perf-test/stable/htmlsingle/#using-environment-variables-as-options

Running a bench mark test:
```
bin/runjava com.rabbitmq.perf.PerfTestMulti publish-consume-spec.js publish-consume-result.js 
```

For documentation on benchmarking see:
https://github.com/rabbitmq/rabbitmq-perf-test/blob/master/html/README.md

### CODE REVIEW

**Comment by rjweeks** [comment]
[SPIKE: performance test rabbits cluster (prod spec) (5 day)](https://trello.com/c/71OfRADc/351-spike-performance-test-rabbits-cluster-prod-spec-5-day)

File: rabbit-perf-test/Makefile [hunk 1]
```diff
@@ -0,0 +1,2 @@
+build:
+	docker build -t eu.gcr.io/census-rm-performance/rm/census-rm-rabbit-performance .
```
Comment (AdamHawtin) [inline rabbit-perf-test/Makefile]: I don't think census-rm-performance is the right project.
Also could be another command in the top level make file instead?

File: rabbit-perf-test/tests/publish-consume-spec.js [hunk 2]
```diff
@@ -0,0 +1,2 @@
+[{'name': 'consume', 'type': 'simple', 'uri': 'amqp://guest:xHwiJyrI7l7LkwOsk9vDL7ph@rabbitmq-discovery.default.svc.cluster.local', 'params':
```
Comment (AdamHawtin) [inline rabbit-perf-test/tests/publish-consume-spec.js]: ⚠️ What is this password?

File: rabbit-perf-test/Makefile (hunk 1 above)
Comment (rjweeks) [reply rabbit-perf-test/Makefile]: Changed to CI.

Comment (rjweeks) [reply rabbit-perf-test/Makefile]: Moved build command to top level make file.

**Comment by rjweeks** [comment]
> Would be a good idea to put how to run the tests in the README

Added.

**Comment by rjweeks** [comment]
@hbrace Thanks for testing. I've made changes that you found testing.

File: rabbit-perf-test/tests/basic-test.sh [hunk 3]
```diff
@@ -0,0 +1 @@
+bin/runjava com.rabbitmq.perf.PerfTest --uri amqp://guest:${RABBITMQ_PASSWORD}@rabbitmq-discovery.default.svc.cluster.local
```
Comment (ryangrundy7) [inline rabbit-perf-test/tests/basic-test.sh]: Needs to be executable

File: README.md [hunk 4]
```diff
@@ -52,5 +52,37 @@ You can alter the environment variables to use different sample files and timing
...
+Set Kubectl to census-rm-performance and then apply the deployment
+```bash
+gcloud beta container clusters get-credentials rm-k8s-cluster --region <REGION> --project <PERFORMANCE_PROJECT_NAME>
+kubectl apply -f rabbit-perf-test/census-rm-rabbit-perf-deployment.yml
+```
+
+Shell into pod to run a basic test.
+This test is continuous so you need to 'CTRL C' to stop it.
+```bash
+./basic-test.sh
```
Comment (ryangrundy7) [inline README.md]: Needs to be updated to the tests directory `./tests/basic-test.sh`

Comment (rjweeks) [reply README.md]: Done.

File: README.md [hunk 5]
```diff
@@ -52,5 +52,37 @@ You can alter the environment variables to use different sample files and timing
...
+```
+
+For documentation on running tests:
+
+https://rabbitmq.github.io/rabbitmq-perf-test/stable/htmlsingle/
+
+Running a bench mark test:
+You need to edit publish-consume-spec.js password for your RabbitMQ cluster.
+```bash
+bin/runjava com.rabbitmq.perf.PerfTestMulti publish-consume-spec.js publish-consume-result.js
```
Comment (ryangrundy7) [inline README.md]: Needs to be updated to the new location in `tests`

File: rabbit-perf-test/census-rm-rabbit-perf-deployment.yml [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+    template:
+      metadata:
+        name: census-rm-rabbit-perf
+        labels:
+          app: census-rm-rabbit-perf
+          env: prod
+      spec:
+        containers:
+        - name: census-rm-rabbit-perf
+          image: eu.gcr.io/census-rm-performance/rm/census-rm-rabbit-performance:latest
```
Comment (ryangrundy7) [inline rabbit-perf-test/census-rm-rabbit-perf-deployment.yml]: Needs to be CI

File: README.md (hunk 5 above)
Comment (ryangrundy7) [reply README.md]: Also need to update this to show to create the spec using the new shell script

Comment (rjweeks) [reply README.md]: Done.

File: README.md [hunk 7]
```diff
@@ -52,5 +52,38 @@ You can alter the environment variables to use different sample files and timing
...
+```
+
+For documentation on running tests:
+
+https://rabbitmq.github.io/rabbitmq-perf-test/stable/htmlsingle/
+
+Running a bench mark test:
+You need to create a benchmark specification file. Please refer to documentation link below. A script has been provided to produce a valid publish-consume-spec.js example specification including the URI for your rabbit cluster.
+```bash
+test/make-benchmark-spec.sh
```
Comment (ryangrundy7) [inline README.md]: This should be `tests` and at the moment, it just outputs it as a string and should pipe it into `ests/publish-consume-spec.js`
```suggestion
./tests/make-benchmark-spec.sh > tests/publish-consume-spec.js
```

File: README.md (hunk 4 above)
Comment (ryangrundy7) [reply README.md]: Hasn't been done!

Comment (rjweeks) [reply README.md]: Pushed now!

File: rabbit-perf-test/tests/make-benchmark-spec.sh [hunk 8]
```diff
@@ -0,0 +1,4 @@
+cat  << EOF
```
Comment (liamtoozer) [inline rabbit-perf-test/tests/make-benchmark-spec.sh]: Is this meant to be `cat > publish-consume-spec.js << EOF`?

**Pull request merged**